### DISCUSSION TITLE
Getting Internal error: org.eclipse.tycho.core.ee.UnknownEnvironmentException: Unknown OSGi execution environment: 'JavaSE-17'

### DISCUSSION THREAD

**Discussion by 7astro7** [dbeaver/dbeaver#13242] [Q&A] [answered]
Hi all, I'm getting 
```
Internal error: org.eclipse.tycho.core.ee.UnknownEnvironmentException: Unknown OSGi execution environment: 'JavaSE-17' -> [Help 1]
```
when I run 
```
mvn clean package
```

and java version produces:  
```
$ java --version
openjdk 17-ea 2021-09-14
OpenJDK Runtime Environment (build 17-ea+19-Debian-1)
OpenJDK 64-Bit Server VM (build 17-ea+19-Debian-1, mixed mode, sharing)
```

I'm unsure how to solve this and assume someone has dealt with this before, thanks for your help

**Comment by mcpiroman** [comment, marked as answer]
Looks like Java 17 is not supported by the build-chain. I think latest supported is 15; 11 definitely works.

**Reply by flambchap** [reply]
Yep, it is a limitation of the current version of Tycho.

**Reply by h3o66** [reply]
Looks like with a bump to version to 2.7.1 it builds on Java 17 - not sure if this ok to just bump it 😅 

I needed this for a Fedora 36 Build:
https://copr.fedorainfracloud.org/coprs/birkch/dbeaver/build/4282059/

Are there any plans to support Java 17 ?
